src/probnum/filtsmooth/gaussfiltsmooth/kalman_utils.py

```diff
@@ -0,0 +1,215 @@
+"""Prediction, update, and smoothing implementations for the Kalman filter/smoother."""
+
+import typing
+
+import numpy as np
+
+import probnum.random_variables as pnrv
+
+from .stoppingcriterion import StoppingCriterion
+
+########################################################################################################################
+# Prediction choices
+########################################################################################################################
+
+
+def predict_via_transition(
+    dynamics_model,
+    start,
+    stop,
+    rv,
+    _intermediate_step=None,
+    _linearise_at=None,
+    _diffusion=1.0,
```

Comment:
here, one can add a square-root implementation (for linear(ised) systems). It needs the same signature, but other than that, anything is possible.